### PR TITLE
Add accounts endpoints

### LINKED ISSUE #39: Create API endpoints for adding, updating, and deleting a social media account.
The following are the endpoints required for modifying users' social media accounts.
- `POST /account/profile/add`: The body should contain the name of the platform and the link the user wants to display along with the account type.
- `POST /account/profile/delete`: The body should contain the ID of this social media platform entry in the `user_accounts` table.
- `POST /account/profile/update`: The body should contain the ID of this social media platform entry in the `user_accounts` table, the updated platform name, the updated link the user wants to display, and the updated account type. This means that the form for editing social media accounts must be generated dynamically, and each HTML `name` attribute must be of the form. For example, for the row that edits the `user_acocunts` entry with id 7, the HTML `name` attributes will be:
  - `label-3`
  - `value-3`
  - `type-3`

The frontend will need to set up forms on the edit profiles page in order to facilitate sending the right data to the right endpoint.

### PR DESCRIPTION
**Pull request by paytonshaltis**
Closes #39 and #68.

This PR implements all of the needed API endpoints for manipulating `user_account` entries in the database. An endpoint for each of the following actions has been set up:

- Adding a new user_account (expecting a label, value, and type).
- Updating a new user_account (expecting a label, value, and type).
- Deleting an existing user_account (expecting an id).
- Getting all user_account entries for a user (expecting an email).

Additionally, the Profile page is fully functional. The user is able to add a new account, edit individual fields of an existing account, as well as delete an existing account. If a user tries editing an account with a label or value that is blank, the previous value will be restored to the text input.

I know that Nisha recently added account types to this page in her PR #81, so there will need to be a merge at some point. I just have buttons that toggle, she has switches: I don't mind what gets used, but I just made something simple here so that I could actually connect the front and back ends together. I would recommend merging this PR first, since there it includes a lot of the functionality for actually modifying the database. Once this PR is in, we can merge Nisha's changes to make sure that they don't conflict with any of the working functionality.

I recommend you try running this for yourself to test the functionality on your own machines.

### CODE REVIEW

File: views/account/profile/main.js
Comment (sderacy) [inline views/account/profile/main.js:61]: I wonder if this is intuitive enough for the user to know that this is how you change the type. There should be some indicator to let them know that they can tap this to change it

**Review by sderacy — COMMENTED** [review]
Looks good, but I'm not necessarily sold on the button

Let me know what you think

**Comment by paytonshaltis** [comment]
> Looks good, but I'm not necessarily sold on the button

As per the description of the pull request, buttons or toggle switches are totally fine to switch out. They were only implemented so that I could actually finish connecting the front and back ends:

>I just have buttons that toggle, she has switches: I don't mind what gets used, but I just made something simple here so that I could actually connect the front and back ends together.

**Comment by paytonshaltis** [comment]
I would, however, merge this PR as is _with_ the buttons, since Nisha's PR implements the toggles. Once this PR is merged, we can deal with the merge conflicts from her request which will replace my buttons for her toggle switches.

**Comment by sderacy** [comment]
I'm aware but since you said:
> I don't mind what gets used

I couldn't tell if the plan was to choose one over the other eventually. I do think these look a little nicer but if we went with this approach, we'd need something explicit to indicate that those are toggles. This obviously isn't enough to make me not approve this but I wanted to make sure you saw this and provided your input on it as well.
 
> I would, however, merge this PR as is _with_ the buttons, since Nisha's PR implements the toggles. Once this PR is merged, we can deal with the merge conflicts from her request which will replace my buttons for her toggle switches.

Sounds good to me, I think waiting until the conflict is done is the best approach to make the decision but I'm ready to merge your PR if you are.

Nice work! :)

**Review by sderacy — APPROVED** [review]
Ship it

**Pull request merged**
